blocksync/buffer.go

```diff
@@ -78,15 +75,12 @@ func (b *blockBuffer) Flush(blk *block.Block) (bool, bCheckinResult) {
 		}
 		delete(b.blocks, heightToSync)
 		if err := commitBlock(b.bc, b.cs, blk); err != nil && errors.Cause(err) != blockchain.ErrInvalidTipHeight {
-			if errors.Cause(err) == poll.ErrProposedDelegatesLength || errors.Cause(err) == poll.ErrDelegatesNotAsExpected || errors.Cause(err) == db.ErrNotExist {
-				l.Debug("Failed to commit the block.", zap.Error(err), zap.Uint64("syncHeight", heightToSync))
-			} else {
-				l.Error("Failed to commit the block.", zap.Error(err), zap.Uint64("syncHeight", heightToSync))
-			}
+			l.With(
+				zap.Uint64("syncHeight", heightToSync),
+				zap.Strings("actionHash", blk.ActionHashs())).Error("Failed to commit the block.", zap.Error(err))
```

Comment:
this if{} is added in https://github.com/iotexproject/iotex-core/commit/2bf61cdd85789412055cd529f49a8db997321e99
looks like to suppress a specific non-critical issue in the past

verified with latest code, this error no longer happens, so remove it and use l.Error() for any error returned by commitBlock()

Reply:
Suggest do not touch this snippet of code if not necessary. You didn't see it perhaps b/c it is `Debug` log which is not enabled.